src/t8_schemes/t8_default/t8_default_pyramid/t8_dpyramid_bits.c

```diff
@@ -1554,34 +1554,39 @@ t8_dpyramid_vertex_reference_coords (const t8_dpyramid_t *elem, const int vertex
 }
 
 void
-t8_dpyramid_compute_reference_coords (const t8_dpyramid_t *elem, const double *ref_coords, double *out_coords)
+t8_dpyramid_compute_reference_coords (const t8_dpyramid_t *elem, const double *ref_coords, const size_t num_coords,
+                                      double *out_coords)
 {
   T8_ASSERT (ref_coords != NULL);
   T8_ASSERT (t8_dpyramid_is_valid (elem));
   if (t8_dpyramid_shape (elem) == T8_ECLASS_PYRAMID) {
     const t8_dpyramid_coord_t length = T8_DPYRAMID_LEN (elem->pyramid.level);
-    out_coords[0] = elem->pyramid.x;
-    out_coords[1] = elem->pyramid.y;
-    out_coords[2] = elem->pyramid.z;
-
-    if (elem->pyramid.type == T8_DPYRAMID_FIRST_TYPE) {
-      out_coords[0] += ref_coords[0] * length;
-      out_coords[1] += ref_coords[1] * length;
-      out_coords[2] += ref_coords[2] * length;
-    }
-    else {
-      out_coords[0] += (ref_coords[0] - ref_coords[2]) * length;
-      out_coords[1] += (ref_coords[1] - ref_coords[2]) * length;
-      out_coords[2] += (1 - ref_coords[2]) * length;
-    }
 
-    /*scale the coordinates onto the reference cube */
-    out_coords[0] /= (double) T8_DPYRAMID_ROOT_LEN;
-    out_coords[1] /= (double) T8_DPYRAMID_ROOT_LEN;
-    out_coords[2] /= (double) T8_DPYRAMID_ROOT_LEN;
+    for (size_t coord = 0; coord < num_coords; ++coord) {
+      const size_t offset = coord * 3;
+      out_coords[offset + 0] = elem->pyramid.x;
+      out_coords[offset + 1] = elem->pyramid.y;
+      out_coords[offset + 2] = elem->pyramid.z;
+
+      if (elem->pyramid.type == T8_DPYRAMID_FIRST_TYPE) {
+        out_coords[offset + 0] += ref_coords[offset + 0] * length;
+        out_coords[offset + 1] += ref_coords[offset + 1] * length;
+        out_coords[offset + 2] += ref_coords[offset + 2] * length;
+      }
+      else {
+        out_coords[offset + 0] += (ref_coords[offset + 0] - ref_coords[offset + 2]) * length;
+        out_coords[offset + 1] += (ref_coords[offset + 1] - ref_coords[offset + 2]) * length;
+        out_coords[offset + 2] += (1 - ref_coords[offset + 2]) * length;
+      }
+
+      /*scale the coordinates onto the reference cube */
+      out_coords[offset + 0] /= (double) T8_DPYRAMID_ROOT_LEN;
+      out_coords[offset + 1] /= (double) T8_DPYRAMID_ROOT_LEN;
+      out_coords[offset + 2] /= (double) T8_DPYRAMID_ROOT_LEN;
+    }
```

Comment:
Please extract the if/else part from the loop

Reply:
Done